pkg/ingester/active_series.go

```diff
@@ -21,8 +21,9 @@ const activeSeriesMaxSizeBytes = 1 * 1024 * 1024
 
 // ActiveSeries implements the ActiveSeries RPC. It returns a stream of active
 // series that match the given matchers.
-func (i *Ingester) ActiveSeries(request *client.ActiveSeriesRequest, stream client.Ingester_ActiveSeriesServer) error {
-	if err := i.checkRunning(); err != nil {
+func (i *Ingester) ActiveSeries(request *client.ActiveSeriesRequest, stream client.Ingester_ActiveSeriesServer) (asErr error) {
+	defer func() { asErr = i.handleReadError(asErr) }()
```

Comment:
What is this supposed to do? `asErr` is nil when this function executes.

Maybe you mean to name the return error `err` instead, and assign to it during the function execution?

Reply:
If there's a naked return, `asErr` is still assigned (to whatever it last value was), as the function inevitably has a return param.

However, I agree that it should be called `err`, not `asErr` :thinking:

Reply:
I will fix this and all other occurrences of the same problem. Thank you for the suggestion.

Reply:
> If there's a naked return

@colega but `asErr` wasn't assigned to in the method, so it would be `nil` in the case of a naked return (since the method would instead assign to `err`). Or am I missing something?